containers/src/Kokkos_DynamicView.hpp

```diff
@@ -685,57 +738,64 @@ inline std::enable_if_t<
               T, P...>::HostMirror::data_type>::value),
     typename Kokkos::Experimental::DynamicView<T, P...>::HostMirror>
 create_mirror_view(const Kokkos::Experimental::DynamicView<T, P...>& src,
-                   const I&... arg_prop) {
-  return Kokkos::create_mirror(arg_prop..., src);
+                   const Impl::ViewCtorProp<ViewCtorArgs...>& arg_prop) {
+  return Kokkos::create_mirror(arg_prop, src);
 }
 
-template <class Space, class T, class... P, class... I>
+template <class Space, class T, class... P, class... ViewCtorArgs>
 inline std::enable_if_t<
     Impl::MirrorDynamicViewType<Space, T, P...>::is_same_memspace,
     typename Kokkos::Impl::MirrorDynamicViewType<Space, T, P...>::view_type>
 create_mirror_view(const Space&,
                    const Kokkos::Experimental::DynamicView<T, P...>& src,
-                   const I&...) {
+                   const Impl::ViewCtorProp<ViewCtorArgs...>&) {
   return src;
 }
 
-template <class Space, class T, class... P, class... I>
+template <class Space, class T, class... P, class... ViewCtorArgs>
 std::enable_if_t<
     !Impl::MirrorDynamicViewType<Space, T, P...>::is_same_memspace,
     typename Kokkos::Impl::MirrorDynamicViewType<Space, T, P...>::view_type>
 create_mirror_view(const Space& space,
                    const Kokkos::Experimental::DynamicView<T, P...>& src,
-                   const I&... arg_prop) {
-  return Kokkos::create_mirror(arg_prop..., space, src);
+                   const Impl::ViewCtorProp<ViewCtorArgs...>& arg_prop) {
+  return create_mirror(space, src, arg_prop);
```

Comment:
did you want to call this here? Or rather add the space to the `arg_prop`?

Reply:
3f5c6c43aafa38aac4c9540d47af14ecd712c2ff